app/forms/idp/edit.tsx

```diff
@@ -104,14 +104,14 @@ export function EditIdpSideModalForm() {
         control={form.control}
         disabled
       />
-      {/* TODO: add group attribute name when it is added to the API
-          <TextField
-            name="groupAttributeName"
-            label="Group attribute name"
-            description="Name of SAML attribute where we can find a comma-separated list of names of groups the user belongs to"
-            control={form.control}
-            disabled
-          /> */}
+      <TextField
+        name="groupAttributeName"
+        label="Group attribute name"
+        description="Name of the SAML attribute sent by the IdP containing a comma-separated list of groups for the authenticated user"
```

Comment:
I don't think they're actually comma-separated. It's all XML.

```xml
<saml:AttributeStatement>
  <saml:Attribute Name="groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
    <saml:AttributeValue>SRE</saml:AttributeValue>
    <saml:AttributeValue>Admins</saml:AttributeValue>
  </saml:Attribute>
</saml:AttributeStatement>
```

https://github.com/oxidecomputer/omicron/blob/5bf05de207bf05201aea7c07f24ddc7724707008/nexus/tests/integration_tests/data/saml_response_with_groups.xml#L94-L98

Maybe we can be a little less concrete about the format anyway. It doesn't matter here:

> Name of the SAML attribute in the IdP response listing the user's groups

We could go even shorter to get it on one line, but I don't think this is consistent with our more typically breezy style.

> Name of SAML attribute in IdP response listing user's groups

Worth noting that `text-balance` looks worse and worse the shorter the thing gets. I'd take it back out.

The docs [also](https://docs.oxide.computer/guides/system/completing-rack-config#_configure_silo_identity_provider) say "comma-separated". I will make a note to confirm it's wrong and fix it.

<img width="877" alt="image" src="https://github.com/user-attachments/assets/46da45e1-a230-48ea-b1ce-38f88765117b">

Reply:
Docs fix: https://github.com/oxidecomputer/docs/pull/484

Reply:
It can be a single AttributeStatement, like you posted, but it can also be multple:

```xml
      <saml:Attribute Name="groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
        <saml:AttributeValue>Admins</saml:AttributeValue>
      </saml:Attribute>
      <saml:Attribute Name="groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
        <saml:AttributeValue>SRE</saml:AttributeValue>
      </saml:Attribute>
```

the Keycloak setting for this is (under the "groups" mapping):

![keycloak_single_group_attribute](https://github.com/user-attachments/assets/a9c5897a-e1bb-48f7-94cf-1c7d870cf944)

It _can_ also be a comma separated list of groups, if the IDP sends that. Some applications apparently do require that (eg: https://support.okta.com/help/s/question/0D54z00009Jau72CAB/group-attribute-statements-combine-all-groups-into-a-comma-separated-string?language=en_US)

Reply:
Very helpful, thanks. Do all of those work the same way with respect to the attribute name? In which case just not mentioning the details seems best.

Reply:
> Do all of those work the same way with respect to the attribute name? In which case just not mentioning the details seems best.

I believe so yeah, both use the same `Attribute Name="groups"`, and it'd be a bug not to. Agree on eliding that detail yeah. Even so if we ever encounter an IDP that only sends comma separated lists Nexus can handle it.